### PR TITLE
libvirt_vm: add set_boot_kernel function

### PR DESCRIPTION
**Pull request by chloerh**

### CODE REVIEW

File: virttest/libvirt_vm.py [hunk 1]
```diff
@@ -1095,13 +1091,32 @@ def set_kernel_param(self, parameter, value=None, remove=False):
...
+        Set a specific kernel parameter.
+
+        :param option: A kernel parameter to set.
+        :param value: The value of the parameter to be set.
+        :param remove: Remove the parameter if True.
+        :return: True if succeed of False if failed.
+        """
+        if self.is_dead():
+            logging.error("Can't set kernel param on a dead VM.")
+            return False
```
Comment (will-Do) [inline virttest/libvirt_vm.py]: As get_grub_file() already check the vm state, so this one can be removed

File: virttest/libvirt_vm.py [hunk 2]
```diff
@@ -1156,6 +1171,59 @@ def set_kernel_param(self, parameter, value=None, remove=False):
...
+        """
+        Set default kernel to the second one or to debug kernel
+        :param index: index of kernel to set to default
+        :param kernel_debug: True if set debug kernel to default
+        :return: default kernel
+        """
+        if self.is_dead():
+            logging.error("Can't set kernel param on a dead VM.")
+            return False
+
```
Comment (will-Do) [inline virttest/libvirt_vm.py]: Same above

File: virttest/libvirt_vm.py [hunk 3]
```diff
@@ -1156,6 +1171,59 @@ def set_kernel_param(self, parameter, value=None, remove=False):
...
+        session = self.wait_for_login()
+        try:
+            grub_path = self.get_grub_file()
+            if not grub_path:
+                return
+            if "grub2" in grub_path:
+                grub = 2
+                cmd_list_all = "cat /boot/grub2/grub.cfg |grep menuentry"
```
Comment (will-Do) [inline virttest/libvirt_vm.py]: Can we get this file by get_grub_file()?

File: virttest/libvirt_vm.py [hunk 4]
```diff
@@ -1156,6 +1171,59 @@ def set_kernel_param(self, parameter, value=None, remove=False):
...
+                out = session.cmd(cmd_list_all)
+                kernel_lst = re.findall("menuentry '.*?'", out)
```
Comment (will-Do) [inline virttest/libvirt_vm.py]: Please update the variable name

File: virttest/libvirt_vm.py [hunk 5]
```diff
@@ -1156,6 +1171,59 @@ def set_kernel_param(self, parameter, value=None, remove=False):
...
+            else:
+                grub = 1
+                out = session.cmd("cat /boot/grub/menu.lst |grep initramfs")
+                kernel_lst = re.findall("-.*", out)
+            if index >= len(kernel_lst):
+                logging.error("Index out of kernel list")
+                return
+            logging.info("kernel list of vm:")
```
Comment (will-Do) [inline virttest/libvirt_vm.py]: Useless log

File: virttest/libvirt_vm.py [hunk 6]
```diff
@@ -1156,6 +1171,59 @@ def set_kernel_param(self, parameter, value=None, remove=False):
         finally:
             session.close()
 
+    def set_boot_kernel(self, index, kernel_debug=False):
+        """
+        Set default kernel to the second one or to debug kernel
+        :param index: index of kernel to set to default
+        :param kernel_debug: True if set debug kernel to default
```
Comment (will-Do) [inline virttest/libvirt_vm.py]: So maybe 'debug_kernel' is more readable

File: virttest/libvirt_vm.py [hunk 7]
```diff
@@ -1071,6 +1071,26 @@ def set_kernel_console(self, device, speed=None, remove=False):
         logging.debug("Set kernel params for %s failed.", device)
         return False
 
+    def get_grub_file(self, session):
+        """
+        Find grub file of vm.
+        :return: grub file path, None if no grub file found
```
Comment (will-Do) [inline virttest/libvirt_vm.py]: It's better to complete the docstring

**Comment by will-Do** [comment]
No more comment from me. @waynesun09 @Hao-Liu would you give ACK for this pr then we can merge it.

File: virttest/libvirt_vm.py [hunk 8]
```diff
@@ -1071,6 +1071,26 @@ def set_kernel_console(self, device, speed=None, remove=False):
         logging.debug("Set kernel params for %s failed.", device)
         return False
 
+    def get_grub_file(self, session):
+        """
+        Find grub file of vm.
```
Comment (waynesun09) [inline virttest/libvirt_vm.py]: For docstring, first line is the description, then you need one empty line following, then details on param and return.

**Comment by chloerh** [comment]
@will-Do @waynesun09 updated as advised, please review, thanks again.

**Pull request merged**